### PR TITLE
Updated to run 2017 MC/DATA

### PR DESCRIPTION
**Pull request by diogodebastos**

### CODE REVIEW

File: TTHAnalysis/cfg/run_susyStop4Body_cfg.py [hunk 1]
```diff
@@ -230,6 +238,9 @@ def jetLepRatio( jet, lepton):
     jetAnaScaleDown.cleanJetsFromTaus = True
 
 isoTrackAna.setOff = False
+
+isoTrackAna.useLegacy2016 = True
```
Comment (cbeiraod) [inline TTHAnalysis/cfg/run_susyStop4Body_cfg.py]: This should be in a if...
If you are analyzing 2017(18) data, you do not want to use the legacy version.

File: TTHAnalysis/python/analyzers/ntupleTypes.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/env python
 from math import *
-from PhysicsTools.Heppy.analyzers.core.autovars import NTupleObjectType  
+from PhysicsTools.Heppy.analyzers.core.autovars import NTupleObjectType
```
Comment (cbeiraod) [inline TTHAnalysis/python/analyzers/ntupleTypes.py]: You shouldn't do any changes on this file... Remember that we want to make changes only to the Stop4Body specific parts and _avoid_ changes to parts not specific (sometimes it is not possible).  Revert all changes on this file. It is mostly from spaces that your editor automatically removed from the end of lines.

**Review by cbeiraod — CHANGES_REQUESTED** [review]

File: TTHAnalysis/python/analyzers/ntupleTypes.py [hunk 3]
```diff
@@ -105,7 +105,7 @@
     NTupleVariable("jetNDau",    lambda lepton : lepton.jet.numberOfDaughters() if hasattr(lepton,'jet') and lepton.jet != lepton else -1, help="n daughters of nearest jet"),
     NTupleVariable("jetNDauCharged",    lambda lepton : sum(x.charge()!=0 for x in lepton.jet.daughterPtrVector()) if hasattr(lepton,'jet') and lepton.jet != lepton else -1, help="n charged daughters of nearest jet"),
     NTupleVariable("jetNDauPV",    lambda lepton : sum(x.charge()!=0 and x.fromPV()==3 for x in lepton.jet.daughterPtrVector()) if hasattr(lepton,'jet') and lepton.jet != lepton else -1, help="n charged daughters from PV of nearest jet"),
-    NTupleVariable("jetNDauNotPV",    lambda lepton : sum(x.charge()!=0 and x.fromPV()<=2 for x in lepton.jet.daughterPtrVector()) if hasattr(lepton,'jet') and lepton.jet != lepton else -1, help="n charged daughters from PV of nearest jet"),    
+    NTupleVariable("jetNDauNotPV",    lambda lepton : sum(x.charge()!=0 and x.fromPV()<=2 for x in lepton.jet.daughterPtrVector()) if hasattr(lepton,'jet') and lepton.jet != lepton else -1, help="n charged daughters from PV of nearest jet"),
```
Comment (cbeiraod) [inline TTHAnalysis/python/analyzers/ntupleTypes.py]: What changed here?

File: TTHAnalysis/python/analyzers/ntupleTypes.py [hunk 4]
```diff
@@ -198,7 +198,7 @@
     NTupleVariable("nSubJets30", lambda x : getattr(x, "nSubJets30", 0), int, help="Number of subjets with pt > 30 (kt, R=0.2)"), 
     NTupleVariable("nSubJets40", lambda x : getattr(x, "nSubJets40", 0), int, help="Number of subjets with pt > 40 (kt, R=0.2)"), 
     NTupleVariable("nSubJetsZ01", lambda x : getattr(x, "nSubJetsZ01", 0), int, help="Number of subjets with pt > 0.1 * pt(jet) (kt, R=0.2)"), 
-    # --------------- 
+    # ---------------
```
Comment (cbeiraod) [inline TTHAnalysis/python/analyzers/ntupleTypes.py]: You forgot to undo your changes here

File: TTHAnalysis/python/analyzers/ntupleTypes.py [hunk 5]
```diff
@@ -227,7 +227,7 @@
 ##------------------------------------------  
 ## MET
 ##------------------------------------------  
-  
+
```
Comment (cbeiraod) [inline TTHAnalysis/python/analyzers/ntupleTypes.py]: And you forgot to undo your changes here too

**Review by cbeiraod — CHANGES_REQUESTED** [review]

File: TTHAnalysis/python/analyzers/ntupleTypes.py (hunk 3 above)
Comment (diogodebastos) [reply TTHAnalysis/python/analyzers/ntupleTypes.py]: I just added whitespaces to match the 94X_dev branch version. But I did it in the wrong place, so I'll fixe it.

File: TTHAnalysis/python/analyzers/ntupleTypes.py (hunk 4 above)
Comment (diogodebastos) [reply TTHAnalysis/python/analyzers/ntupleTypes.py]: :+1:

File: TTHAnalysis/python/analyzers/ntupleTypes.py (hunk 5 above)
Comment (diogodebastos) [reply TTHAnalysis/python/analyzers/ntupleTypes.py]: :+1:

**Pull request merged**